src/yb/tools/yb-admin_client.cc

```diff
@@ -1738,10 +1738,34 @@ Status ClusterAdminClient::ModifyPlacementInfo(
   master::PlacementInfoPB* live_replicas = new master::PlacementInfoPB;
   live_replicas->set_num_replicas(replication_factor);
 
+  int total_min_replica_count = 0;
+
   // Iterate over the placement blocks of the placementInfo structure.
   std::unordered_map<std::string, int> placement_to_min_replicas;
-  for (int iter = 0; iter < placement_info_split.size(); iter++) {
-    placement_to_min_replicas[placement_info_split[iter]]++;
+  for (const auto& placement_block : placement_info_split) {
+    std::vector<std::string> placement_info_min_replica_split = strings::Split(
+        placement_block, ":", strings::AllowEmpty());
+
+    if(placement_info_min_replica_split.size() == 0 ||
+        placement_info_min_replica_split.size() > 2) {
+      return STATUS(InvalidCommand, "Each placement info must have at most 2 values separated by a colon. "
+          "Format: cloud.region.zone:[min_replica_count]. Invalid placement info: " + placement_block);
+    }
+
+    std::string placement_target = placement_info_min_replica_split[0],
+        placement_min_replica_count = "1";
```

Comment:
Let's keep this variable of integer type and perform the type conversion in the `if` block below.